From 3e30e6dfcac336e35ec53b843903ea9818fd0cce Mon Sep 17 00:00:00 2001
From: Karolis Mituzas <karolis.mituzas@imgtec.com>
Date: Mon, 19 Jun 2023 11:30:54 +0100
Subject: arm64: config: Add Imagination-specific config fragment

This fragment, overlaid on arm64 defconfig, produces a kernel that works
on MT8173 (Chromebook).

To use the fragment, the fragment name should be added to the
configuration line, so e.g.
  make ARCH=arm64 defconfig
becomes
  make ARCH=arm64 defconfig mt8173_img_fragment.config
---
 arch/arm64/configs/mt8173_img_fragment.config | 297 ++++++++++++++++++
 1 file changed, 297 insertions(+)
 create mode 100644 arch/arm64/configs/mt8173_img_fragment.config

diff --git a/arch/arm64/configs/mt8173_img_fragment.config b/arch/arm64/configs/mt8173_img_fragment.config
new file mode 100644
index 000000000000..f2801715114d
--- /dev/null
+++ b/arch/arm64/configs/mt8173_img_fragment.config
@@ -0,0 +1,297 @@
+# Arch trims
+CONFIG_ARCH_ACTIONS=n
+CONFIG_ARCH_ALPINE=n
+CONFIG_ARCH_APPLE=n
+CONFIG_ARCH_BCM=n
+CONFIG_ARCH_BERLIN=n
+CONFIG_ARCH_EXYNOS=n
+CONFIG_ARCH_HISI=n
+CONFIG_ARCH_INTEL_SOCFPGA=n
+CONFIG_ARCH_K3=n
+CONFIG_ARCH_KEEMBAY=n
+CONFIG_ARCH_LG1K=n
+CONFIG_ARCH_MESON=n
+CONFIG_ARCH_MVEBU=n
+CONFIG_ARCH_NPCM=n
+CONFIG_ARCH_NXP=n
+CONFIG_ARCH_QCOM=n
+CONFIG_ARCH_RENESAS=n
+CONFIG_ARCH_ROCKCHIP=n
+CONFIG_ARCH_SEATTLE=n
+CONFIG_ARCH_SPRD=n
+CONFIG_ARCH_SUNXI=n
+CONFIG_ARCH_SYNQUACER=n
+CONFIG_ARCH_TEGRA=n
+CONFIG_ARCH_THUNDER=n
+CONFIG_ARCH_THUNDER2=n
+CONFIG_ARCH_UNIPHIER=n
+CONFIG_ARCH_VEXPRESS=n
+CONFIG_ARCH_VISCONTI=n
+CONFIG_ARCH_XGENE=n
+CONFIG_ARCH_ZYNQMP=n
+
+# DRM trim
+CONFIG_DRM_CDNS_MHDP8546=n
+CONFIG_DRM_ETNAVIV=n
+CONFIG_DRM_HDLCD=n
+CONFIG_DRM_HISI_KIRIN=n
+CONFIG_DRM_I2C_ADV7511_AUDIO=n
+CONFIG_DRM_KOMEDA=n
+CONFIG_DRM_LEGACY=n
+CONFIG_DRM_LIMA=n
+CONFIG_DRM_LONTIUM_LT8912B=n
+CONFIG_DRM_LONTIUM_LT9611=n
+CONFIG_DRM_LONTIUM_LT9611UXC=n
+CONFIG_DRM_NWL_MIPI_DSI=n
+CONFIG_DRM_PANEL_BOE_TV101WUM_NL6=n
+CONFIG_DRM_PANEL_MANTIX_MLAF057WE51=n
+CONFIG_DRM_PANEL_RAYDIUM_RM67191=n
+CONFIG_DRM_PANEL_SITRONIX_ST7703=n
+CONFIG_DRM_PANFROST=n
+CONFIG_DRM_PL111=n
+CONFIG_DRM_RCAR_DW_HDMI=n
+CONFIG_DRM_RCAR_USE_LVDS=n
+CONFIG_DRM_RCAR_USE_MIPI_DSI=n
+CONFIG_DRM_SIMPLE_BRIDGE=n
+CONFIG_DRM_TIDSS=n
+CONFIG_DRM_MXSFB=n
+
+# Explicitly needed
+CONFIG_BACKLIGHT_LP855X=y
+CONFIG_BACKLIGHT_PWM=y
+CONFIG_CFG80211=y
+CONFIG_CIFS=m
+CONFIG_NFS_FS=m
+CONFIG_NFS_SWAP=y
+CONFIG_NFS_V2=m
+CONFIG_NFS_V3=m
+CONFIG_NFS_V3_ACL=y
+CONFIG_NFS_V4=m
+CONFIG_NFS_V4_1=n
+CONFIG_COMMON_CLK_MT6797_IMGSYS=y
+CONFIG_COMMON_CLK_MT6797_MMSYS=y
+CONFIG_COMMON_CLK_MT6797_VDECSYS=y
+CONFIG_COMMON_CLK_MT6797_VENCSYS=y
+CONFIG_CPU_FREQ_GOV_CONSERVATIVE=y
+CONFIG_CPU_FREQ_GOV_POWERSAVE=y
+CONFIG_CPU_IDLE_GOV_LADDER=y
+CONFIG_DRM=y
+CONFIG_DRM_PANEL_EDP=y
+CONFIG_DRM_PARADE_PS8640=y
+CONFIG_SND_SOC_MT8173_RT5650_RT5676=y
+CONFIG_FB_SIMPLE=y
+
+# Explicitly disabled
+CONFIG_BLK_DEBUG_FS=n
+
+# Not reviewed
+CONFIG_DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT=n
+CONFIG_DEBUG_INFO_NONE=y
+CONFIG_DEVFREQ_GOV_USERSPACE=n
+CONFIG_DP83867_PHY=n
+CONFIG_DRM_ANALOGIX_ANX78XX=y
+CONFIG_DRM_ANALOGIX_DP=y
+CONFIG_DRM_DISPLAY_CONNECTOR=y
+CONFIG_DRM_DISPLAY_HELPER=y
+CONFIG_DRM_DP_AUX_BUS=y
+CONFIG_DRM_GEM_DMA_HELPER=y
+CONFIG_DRM_KMS_HELPER=y
+CONFIG_DRM_MEDIATEK=y
+CONFIG_DRM_MEDIATEK_HDMI=y
+CONFIG_DW_WATCHDOG=n
+CONFIG_ENERGY_MODEL=n
+CONFIG_ETHERNET=n
+CONFIG_EXTCON_PTN5150=n
+CONFIG_EXTCON_USBC_CROS_EC=n
+CONFIG_FAILOVER=n
+CONFIG_FW_LOADER_USER_HELPER_FALLBACK=y
+CONFIG_GRACE_PERIOD=m
+CONFIG_HIBERNATION=n
+CONFIG_I2C_CADENCE=n
+CONFIG_I2C_CROS_EC_TUNNEL=n
+CONFIG_I2C_HID_OF=n
+CONFIG_IIO_CROS_EC_SENSORS_CORE=n
+CONFIG_IIO_ST_LSM6DSX=n
+CONFIG_IIO_ST_MAGN_3AXIS=n
+CONFIG_INET_UDP_DIAG=y
+CONFIG_INPUT_PWM_BEEPER=n
+CONFIG_INPUT_PWM_VIBRA=n
+CONFIG_INTERCONNECT=n
+CONFIG_IOMMU_IO_PGTABLE_DART=n
+CONFIG_KEXEC_FILE=n
+CONFIG_LEDS_CLASS_MULTICOLOR=n
+CONFIG_LEDS_LM3692X=n
+CONFIG_LEDS_PCA9532=n
+CONFIG_LOCKD=m
+CONFIG_LOGO=n
+CONFIG_MAC80211=y
+CONFIG_MAC80211_LEDS=n
+CONFIG_MACVLAN=n
+CONFIG_MARVELL_10G_PHY=n
+CONFIG_MARVELL_PHY=n
+CONFIG_MDIO_BCM_UNIMAC=n
+CONFIG_MDIO_BITBANG=n
+CONFIG_MDIO_BUS_MUX_MMIOREG=n
+CONFIG_MDIO_BUS_MUX_MULTIPLEXER=n
+CONFIG_MEDIA_PLATFORM_SUPPORT=n
+CONFIG_MEDIA_SUPPORT=y
+CONFIG_MEMORY_HOTPLUG=n
+CONFIG_MEMTEST=n
+CONFIG_MFD_MT6360=n
+CONFIG_MFD_WM8994=n
+CONFIG_MHI_BUS=n
+CONFIG_MICREL_PHY=n
+CONFIG_MICROSEMI_PHY=n
+CONFIG_MII=m
+CONFIG_MMC_SDHCI_AM654=n
+CONFIG_MMC_SDHCI_OF_DWCMSHC=n
+CONFIG_MOUSE_ELAN_I2C=y
+CONFIG_MQ_IOSCHED_KYBER=n
+CONFIG_MTD_NAND_BRCMNAND=n
+CONFIG_MTK_CMDQ=y
+CONFIG_MTK_CMDQ_MBOX=y
+CONFIG_MTK_CQDMA=y
+CONFIG_MTK_DEVAPC=n
+CONFIG_NETFILTER=n
+CONFIG_NET_9P=n
+CONFIG_NET_DSA=n
+CONFIG_NET_FAILOVER=n
+CONFIG_NET_SCHED=n
+CONFIG_NET_SWITCHDEV=n
+CONFIG_NFC=n
+CONFIG_NVMEM_MTK_EFUSE=n
+CONFIG_NVMEM_RMEM=n
+CONFIG_OF_OVERLAY=y
+CONFIG_OVERLAY_FS=y
+CONFIG_OVERLAY_FS_INDEX=y
+CONFIG_OVERLAY_FS_METACOPY=y
+CONFIG_OVERLAY_FS_REDIRECT_DIR=y
+CONFIG_OVERLAY_FS_XINO_AUTO=y
+CONFIG_PAGE_REPORTING=n
+CONFIG_PCI=n
+CONFIG_PHYLINK=m
+CONFIG_PHY_CADENCE_SIERRA=n
+CONFIG_PHY_CADENCE_TORRENT=n
+CONFIG_PHY_MTK_HDMI=y
+CONFIG_PHY_MTK_MIPI_DSI=y
+CONFIG_PHY_QCOM_USB_HS=y
+CONFIG_PM_DEBUG=y
+CONFIG_POSIX_MQUEUE=n
+CONFIG_PPS=n
+CONFIG_PTP_1588_CLOCK=n
+CONFIG_PWM_CROS_EC=n
+CONFIG_PWM_MEDIATEK=y
+CONFIG_PWM_MTK_DISP=y
+CONFIG_QRTR=n
+CONFIG_RC_CORE=n
+CONFIG_REALTEK_PHY=n
+CONFIG_REGULATOR_DA9211=y
+CONFIG_REGULATOR_DEBUG=y
+CONFIG_REGULATOR_MP8859=n
+CONFIG_REGULATOR_MT6358=n
+CONFIG_REGULATOR_MT6359=n
+CONFIG_REGULATOR_PCA9450=n
+CONFIG_REGULATOR_PF8X00=n
+CONFIG_REGULATOR_TPS65132=n
+CONFIG_RELAY=y
+CONFIG_RFKILL=n
+CONFIG_ROCKCHIP_PHY=n
+CONFIG_RPMSG_QCOM_GLINK_RPM=n
+CONFIG_SENSORS_GPIO_FAN=n
+CONFIG_SENSORS_JC42=n
+CONFIG_SENSORS_LM75=n
+CONFIG_SLAB=y
+CONFIG_SLIMBUS=n
+CONFIG_SLUB=n
+CONFIG_SOC_BRCMSTB=n
+CONFIG_SOUNDWIRE=n
+CONFIG_SPI_BITBANG=y
+CONFIG_SPI_DESIGNWARE=n
+CONFIG_SPI_GPIO=y
+CONFIG_SPI_MT65XX=y
+CONFIG_SPI_SPIDEV=y
+CONFIG_SPMI=n
+CONFIG_STAGING=n
+CONFIG_STM=n
+CONFIG_STRICT_DEVMEM=n
+CONFIG_SUNRPC=m
+CONFIG_SUNRPC_GSS=m
+CONFIG_SUSPEND=n
+CONFIG_TASKSTATS=n
+CONFIG_TI_ADS1015=n
+CONFIG_TOUCHSCREEN_EDT_FT5X06=n
+CONFIG_TOUCHSCREEN_ELAN=y
+CONFIG_TOUCHSCREEN_GOODIX=n
+CONFIG_TYPEC_HD3SS3220=n
+CONFIG_TYPEC_TCPCI=n
+CONFIG_TYPEC_TPS6598X=n
+CONFIG_UACCE=n
+CONFIG_UNIX_DIAG=y
+CONFIG_USB_ALI_M5632=y
+CONFIG_USB_AN2720=y
+CONFIG_USB_CATC=m
+CONFIG_USB_CDNS_SUPPORT=n
+CONFIG_USB_CONFIGFS=n
+CONFIG_USB_EPSON2888=y
+CONFIG_USB_ETH=m
+CONFIG_USB_IPHETH=m
+CONFIG_USB_KAWETH=m
+CONFIG_USB_KC2190=y
+CONFIG_USB_NET_CDC_EEM=m
+CONFIG_USB_NET_CDC_MBIM=m
+CONFIG_USB_NET_CH9200=m
+CONFIG_USB_NET_CX82310_ETH=m
+CONFIG_USB_NET_GL620A=m
+CONFIG_USB_NET_HUAWEI_CDC_NCM=m
+CONFIG_USB_NET_INT51X1=m
+CONFIG_USB_NET_KALMIA=m
+CONFIG_USB_NET_QMI_WWAN=m
+CONFIG_USB_NET_RNDIS_HOST=m
+CONFIG_USB_NET_SR9700=m
+CONFIG_USB_SERIAL=y
+CONFIG_USB_SERIAL_CP210X=n
+CONFIG_USB_SERIAL_FTDI_SIO=n
+CONFIG_USB_SERIAL_OPTION=n
+CONFIG_USB_SERIAL_PL2303=y
+CONFIG_USB_SIERRA_NET=m
+CONFIG_USB_VL600=m
+CONFIG_USB_WDM=m
+CONFIG_USB_XHCI_PCI_RENESAS=n
+CONFIG_VCNL4000=n
+CONFIG_VFIO=n
+CONFIG_VIDEO_DEV=y
+CONFIG_VIDEO_IMX219=n
+CONFIG_VIDEO_OV5640=n
+CONFIG_VIDEO_OV5645=n
+CONFIG_VIRTIO_MENU=n
+CONFIG_VIRTIO_NET=n
+CONFIG_VIRTUALIZATION=n
+CONFIG_VITESSE_PHY=n
+CONFIG_VLAN_8021Q=n
+CONFIG_WQ_POWER_EFFICIENT_DEFAULT=n
+CONFIG_PM_ADVANCED_DEBUG=y
+CONFIG_USB_ETH_EEM=y
+CONFIG_USB_SERIAL_CONSOLE=y
+
+# Random tidyup
+CONFIG_AQUANTIA_PHY=n
+CONFIG_AT803X_PHY=n
+CONFIG_ATA=n
+CONFIG_BATTERY_MAX17042=n
+CONFIG_BCM54140_PHY=n
+CONFIG_BCM7XXX_PHY=n
+CONFIG_BLK_DEV_INTEGRITY=n
+CONFIG_BROADCOM_PHY=n
+CONFIG_CAN=n
+CONFIG_CHARGER_BQ25890=n
+CONFIG_CHARGER_BQ25980=n
+CONFIG_CLK_VEXPRESS_OSC=n
+CONFIG_XEN=n
+CONFIG_FPGA=n
+CONFIG_SOC_TI=n
+CONFIG_GNSS=n
+CONFIG_BT=n
+
+# Needs further review
+CONFIG_CPU_FREQ_DEFAULT_GOV_PERFORMANCE=y
+CONFIG_DRM_PANEL_SIMPLE=y
-- 
2.41.0

